contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,910 @@
+                     *
+                     * Also sqrtPriceX96 is 0 because the pool is already created
+                     * non zero amount attempts to create a new instance of the pool
+                     */
+                    INonfungiblePositionManager.MintParams({
+                        token0: WETH,
+                        token1: OETHb,
+                        tickSpacing: tickSpacing,
+                        tickLower: lowerTick,
+                        tickUpper: upperTick,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(
+            _currentPrice >= sqrtRatioX96Tick0 &&
```

Comment:
🔴 I have bad vibes on this one. Would have to test to see, but I'm concerned that both sides are =< <=, which means that the edges are included. It feels like one of these ratio edges would be in a different tick, right?

Reply:
Yeah agreed, the function would probably either revert when minting OETH or when `estimatingAmount1`, still it would make it a cryptic revert message. Excluding both tick edges as valid to favour the estimation math later on: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/866a671798892ad75106cde0a161e46663366cde
(excuse the accidental `prettier` in the commit)